tmail-backend/jmap/extensions-cassandra/src/main/scala/com/linagora/tmail/james/jmap/settings/CassandraJmapSettingsRepository.scala

```diff
@@ -0,0 +1,45 @@
+package com.linagora.tmail.james.jmap.settings
+
+import com.google.common.base.Preconditions
+import com.google.inject.multibindings.Multibinder
+import com.google.inject.{AbstractModule, Scopes}
+import jakarta.inject.Inject
+import org.apache.james.backends.cassandra.components.CassandraModule
+import org.apache.james.core.Username
+import org.apache.james.jmap.core.UuidState
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+class CassandraJmapSettingsRepository @Inject()(dao: CassandraJmapSettingsDAO) extends JmapSettingsRepository {
+  override def get(username: Username): Publisher[JmapSettings] = dao.selectOne(username)
+
+  override def reset(username: Username, settings: JmapSettingsUpsertRequest): Publisher[SettingsStateUpdate] = {
+    val newState: UuidState = JmapSettingsStateFactory.generateState()
+    val oldState: UuidState = dao.selectState(username).block()
+    SMono.fromCallable(() => {
+      dao.updateSetting(username, Some(newState), Some(settings.settings))
+    }).`then`(SMono.just(SettingsStateUpdate(oldState, newState)))
+  }
+
+  override def updatePartial(username: Username, settingsPatch: JmapSettingsPatch): Publisher[SettingsStateUpdate] = {
```

Comment:
I have serious doubts on the commutativity of your "updatePartial" method" and strongly believes the contract previously written to be insufficient. (set do not commutes)

Do we have a test:

```
update partial - insert (k1 -> v1)
update partial - insert (k2 -> v2)

Expected result (k1 -> v1 + k2 -> v2)
```

Also concurrent testing can be helpful to check for a valid CRDT (using ConcurrentTestRunner)

```
With 10 thread tx do 10 options oy
foreach do: partial update - insert (kx-y -> vx-y)
Expected result: settings with 100 entries, right values
```

(Cassandra have a specific semantic for maps partial updates that MUST be used here)

Reply:
> Do we have a test:
> 
> ```
> update partial - insert (k1 -> v1)
> update partial - insert (k2 -> v2)
> 
> Expected result (k1 -> v1 + k2 -> v2)
> ```
> 

It alreadys. 
`JmapSettingsRepositoryContract#updatePartialShouldAppendNewSettingEntryInPatch`

Reply:
I think Benoit' idea here is more testing partial update concurrently on different setting entries.